fsharp-backend/src/LibExecution/StdLib/LibDict.fs

```diff
@@ -88,154 +90,57 @@ let fns : List<BuiltInFn> =
-    //                               (Dval.to_developer_repr_v0 k)
-    //                               tipe
-    //                         | DList l ->
-    //                             Printf.sprintf
-    //                               "It has length %i but must have length 2."
-    //                               (List.length l)
-    //                         | non_list ->
-    //                             let tipe =
-    //                               non_list
-    //                               |> Dval.tipe_of
-    //                               |> Dval.tipe_to_developer_repr_v0
-    //                             in
-    //                             Printf.sprintf
-    //                               "It is of type `%s` instead of `List`."
-    //                               tipe
-    //                       in
-    //                       Error
-    //                         (DError
-    //                            ( SourceNone
-    //                            , Printf.sprintf
-    //                                "Expected every value within the `entries` argument passed to `%s` to be a `[key, value]` list. However, that is not the case for the value at index %i: `%s`. %s"
-    //                                state.executing_fnname
-    //                                idx
-    //                                (Dval.to_developer_repr_v0 v)
-    //                                err_details )))
-    //             in
-    //             let result =
-    //               l
-    //               |> List.foldi (Ok DvalMap.empty) fold_fn
-    //               |> Result.map (fun o -> DOption (OptJust (DObj o)))
-    //             in
-    //             (match result with Ok res -> res | Error v -> v)
-    //         | _ -> args
-    //             incorrectArgs ())
-    //   ; sqlSpec = NotYetImplementedTODO
-    //   ; previewable = Pure
-    //   ; deprecated = NotDeprecated }
+    { name = fn "Dict" "fromListOverwritingDuplicates" 0
+      parameters = [Param.make "entries" (TList varA) ""]
+      returnType = TDict varA
+      description =
+        "Returns a new dict with `entries`. Each value in `entries` must be a `[key, value]` list, where `key` is a `String`.
+        If `entries` contains duplicate `key`s, the last entry with that key will be used in the resulting dictionary (use `Dict::fromList` if you want to enforce unique keys).
+        This function is the opposite of `Dict::toList`."
+      fn =
+        (function
+        | state, [DList l] ->
+          let f acc e =
+            match e with
+              | DList [DStr k; value] when Map.containsKey k acc -> Map.remove k acc |> Map.add k value
+              | DList [DStr k; value] -> Map.add k value acc
+              | DList [k; value] -> Errors.throw(Errors.argumentWasnt "String`s" "key" k)
+              | _ -> Errors.throw "All list items must be `[key, value]`"
```

Comment:
There's an error condition in the old code, where if there is a DIncomplete, DErorrRail or DError found, it is returned. You can replicate that here using `Errors.foundFakeDval`. (Please add a test for it too, if possible)